File: src/primes.java
import static java.lang.System.out;

public class primes {

    public static int PRIMES = 5000;

	public static class Primes
	{
		public int prime_count;
		public int[] primes = new int[PRIMES];

		public int getPrimeCount () { return this.prime_count; }
		public int getPrime (int i) { return this.primes[i]; }
		public void addPrime (int p) { this.primes[this.prime_count++] = p; }

		public boolean isPrimeDivisible(int candidate) {
			for (int i = 1; i < this.prime_count; ++i) {
                if (this.primes[i] * this.primes[i] > candidate) return false;
				if ((candidate % this.primes[i]) == 0) return true;
			}
			return false;
		}
	}
	
	/**
	 * @param args
	 */
	public static void main(String[] args) {
        PRIMES = args.length > 0 ? Integer.parseInt(args[0]) : 5000;
		Primes p = new Primes();
		int c = 1;
		while (p.getPrimeCount() < PRIMES) {
			if (!p.isPrimeDivisible(c)) {
				p.addPrime(c);
			}
			c++;
		}
		out.println(p.getPrime(p.getPrimeCount() - 1));
	}

}

File: src/primes.py
import sys
primes = int(sys.argv[1] if len(sys.argv) > 1 else "5000")

class Primes:
    def __init__(self):
        self.prime_count = 0
        self.primes = [0]*primes

    def getPrimeCount(self):
        return self.prime_count

    def getPrime(self, i):
        return self.primes[i]

    def addPrime(self, i):
        self.primes[self.prime_count] = i
        self.prime_count += 1

    def isPrimeDivisible(self, candidate):
        for i in range(1, self.prime_count - 1):
            if (self.primes[i] * self.primes[i] > candidate):
                return False
            if (candidate % self.primes[i]) == 0:
                return True
        return False

p = Primes()
c = 1
while p.getPrimeCount() < primes:
    if not p.isPrimeDivisible(c):
        p.addPrime(c)
    c += 1

print(p.getPrime(p.getPrimeCount()-1))
